devel/libecl/src/ecl_grid.c

```diff
@@ -5717,11 +5721,24 @@ static void ecl_grid_fwrite_EGRID__( ecl_grid_type * grid , fortio_type * fortio
   {
     {
       ecl_grid_assert_coord_kw( grid );
-      ecl_kw_fwrite( grid->coord_kw , fortio );
+      if(is_metric != grid->is_metric){
+          ecl_kw_type * coord_kw = ecl_kw_alloc_data_copy(grid->coord_kw);
```

Comment:
Here you should use the function `ecl_kw_alloc_copy()` which will create a full `ecl_kw_type` instance, the `ecl_kw_alloc_data_copy()` will only copy the internal *data* vector; not the header data.